pyston/data_processor.py

```diff
@@ -108,26 +108,39 @@ def _process_file_data(self, data, files, key, data_item, file_content):
             )
             data[key] = filename
         else:
-            self.errors[key] = ugettext('Unsupported file type')
+            self.errors[key] = RESTValidationError(ugettext(
+                'Content type cannot be evaluated from the filename please send it or change the filename'
+            ))
 
     def _process_file_data_field(self, data, files, key, data_item):
         try:
             file_content = BytesIO(base64.b64decode(data_item.get('content').encode('utf-8')))
             self._process_file_data(data, files, key, data_item, file_content)
-        except TypeError:
-            self.errors[key] = ugettext('File content is not in base64 format')
+        except (TypeError, binascii.Error):
+            self.errors[key] = RESTDictError({'content': RESTValidationError(
+                ugettext('File content must be in base64 format')
+            )})
 
     def _process_file_data_url_field(self, data, files, key, data_item):
         url = data_item.get('url')
         try:
-            file_content = get_file_content_from_url(url, pyston_settings.FILE_SIZE_LIMIT)
-            self._process_file_data(data, files, key, data_item, file_content)
-        except RequestDataTooBig:
-            self.errors[key] = ugettext('Response too large, maximum size is {} bytes').format(
-                pyston_settings.FILE_SIZE_LIMIT
-            )
-        except RequestException:
-            self.errors[key] = ugettext('File is unreachable')
+            url_validator(url)
+        except ValidationError as e:
+            self.errors[key] = RESTDictError({'url': RESTValidationError(e.messages[0])})
+        else:
```

Comment:
zbytečné dej to do try bloku. Nebudeš pak mít try/except v try/except

Reply:
ok